src/Microsoft.AspNet.Hosting/Internal/WebHostOptions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.AspNet.Hosting.Internal
+{
+    public class WebHostOptions
+    {
+        private const string OldEnvironmentKey = "ENV";
+
+        public WebHostOptions()
+        {
+        }
+
+        public WebHostOptions(IConfiguration configuration)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            Application = configuration[WebApplicationConfiguration.ApplicationKey];
+            DetailedErrors = ParseBool(configuration, WebApplicationConfiguration.DetailedErrorsKey);
+            CaptureStartupErrors = ParseBool(configuration, WebApplicationConfiguration.CaptureStartupErrorsKey);
+            Environment = configuration[WebApplicationConfiguration.EnvironmentKey] ?? configuration[OldEnvironmentKey];
+            Server = configuration[WebApplicationConfiguration.ServerKey];
+            WebRoot = configuration[WebApplicationConfiguration.WebRootKey];
+        }
+
+        public string Application { get; set; }
```

Comment:
I don't think `set` is needed if the only place you are setting this property is the constructor.